libcloud/compute/drivers/azure_arm.py

```diff
@@ -1942,39 +1945,40 @@ def _to_node(self, data, fetch_nic=True):
                     pass
 
         state = NodeState.UNKNOWN
-        try:
-            action = "%s/InstanceView" % (data["id"])
-            r = self.connection.request(action,
-                                        params={"api-version": "2015-06-15"})
-            for status in r.object["statuses"]:
-                if status["code"] in ["ProvisioningState/creating"]:
-                    state = NodeState.PENDING
-                    break
-                elif status["code"] == "ProvisioningState/deleting":
-                    state = NodeState.TERMINATED
-                    break
-                elif status["code"].startswith("ProvisioningState/failed"):
-                    state = NodeState.ERROR
-                    break
-                elif status["code"] == "ProvisioningState/updating":
-                    state = NodeState.UPDATING
-                    break
-                elif status["code"] == "ProvisioningState/succeeded":
-                    pass
+        if fetch_status:
+            try:
+                action = "%s/InstanceView" % (data["id"])
+                r = self.connection.request(action,
+                                            params={"api-version": "2015-06-15"})
+                for status in r.object["statuses"]:
+                    if status["code"] in ["ProvisioningState/creating"]:
+                        state = NodeState.PENDING
+                        break
+                    elif status["code"] == "ProvisioningState/deleting":
+                        state = NodeState.TERMINATED
+                        break
+                    elif status["code"].startswith("ProvisioningState/failed"):
+                        state = NodeState.ERROR
+                        break
+                    elif status["code"] == "ProvisioningState/updating":
+                        state = NodeState.UPDATING
+                        break
+                    elif status["code"] == "ProvisioningState/succeeded":
+                        pass
```

Comment:
The ProvisioningState is available from the virtual machines list resource group response: https://docs.microsoft.com/en-us/rest/api/compute/virtualmachines/virtualmachines-list-resource-group
without having to make an extra request.

I think `"provisioningState": "succeeded"` is close enough to return NodeState.RUNNING even without the additional detail of PowerState.